src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -167,14 +176,28 @@ public sealed class CultureAwareComparer : StringComparer
     {
         private readonly CompareInfo _compareInfo; // Do not rename (binary serialization)
         private readonly bool _ignoreCase; // Do not rename (binary serialization)
+        private CompareOptions _compareOptions;
 
         internal CultureAwareComparer(CultureInfo culture, bool ignoreCase)
         {
             _compareInfo = culture.CompareInfo;
             _ignoreCase = ignoreCase;
+            _compareOptions = _ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
         }
 
-        private CompareOptions Options => _ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
+        internal CultureAwareComparer(CultureInfo culture, CompareOptions compareOptions)
+        {
+            _compareInfo = culture.CompareInfo;
+            _compareOptions = compareOptions;
```

Comment:
It's a shame you have to keep _ignoreCase around as it's now redundant, but I agree you can't remove it due to serialization.